mesonbuild/compilers/c.py

```diff
@@ -129,16 +146,17 @@ def get_options(self):
             })
         return opts
 
-    def get_option_compile_args(self, options):
+    def get_option_compile_args(self, options: 'OptionDictType') -> T.List[str]:
         args = []
         std = options['std']
         if std.value != 'none':
             args.append('-std=' + std.value)
         return args
 
-    def get_option_link_args(self, options):
+    def get_option_link_args(self, options: 'OptionDictType') -> T.List[str]:
         if self.info.is_windows() or self.info.is_cygwin():
-            return options['winlibs'].value[:]
+            # without a typedict mypy can't understand this.
+            return T.cast(T.List[str], options['winlibs'].value[:])
```

Comment:
Maybe use something like this here (and in all other instances) instead?

```suggestion
            res = options['winlibs'].value[:]
            assert isinstance(res, list)
            for i in res:
                assert isinstance(i, str)
            return res
```

Or would the impact of the explicit check be too significant here?

Reply:
I'm not sure what the right thing to do is here. I'd really like to use a TypedDict for this, but the way we dynamically construct that list makes it rather difficult to do that.

Reply:
I am also not 100% certain what is better, but with the asserts the types are at least checked and it will blow up immediately if something changes.

Reply:
I went ahead and changed to asserts, it's not a particularly hot path so it shouldn't be a problem to get a little bit more safety.